gcloud-java-storage/src/main/java/com/google/gcloud/storage/StorageImpl.java

```diff
@@ -259,14 +262,17 @@ public BucketInfo apply(com.google.api.services.storage.model.Bucket bucketPb) {
           }
         }, serviceOptions.retryParams(), EXCEPTION_HANDLER);
     String cursor = result.x();
-    return new BaseListResult<>(new BlobPageFetcher(bucket, serviceOptions, cursor, optionsMap), cursor,
-        Iterables.transform(result.y(),
+    Iterable<BlobInfo> blobs =
+        result.y() == null ? ImmutableList.<BlobInfo>of() : Iterables.transform(result.y(),
             new Function<StorageObject, BlobInfo>() {
```

Comment:
Fine as is. Another way (insignificantly less efficient and arguable more readable could have been just to change `result.y()` to `MoreObjects.firstNonNull(result.y(), ImmutableList.of())`, ....